libvalkey/async.c

```diff
@@ -359,129 +359,129 @@ static void __redisRunDisconnectCallback(redisAsyncContext *ac, int status)
-        assert(ret == REDIS_ERR);
+        int ret = valkeyShiftCallback(&ac->replies,NULL);
+        assert(ret == VALKEY_ERR);
     } else {
         /* Disconnection is caused by an error, make sure that pending
          * callbacks cannot call new commands. */
-        c->flags |= REDIS_DISCONNECTING;
+        c->flags |= VALKEY_DISCONNECTING;
     }
 
     /* cleanup event library on disconnect.
      * this is safe to call multiple times */
     _EL_CLEANUP(ac);
 
-    /* For non-clean disconnects, __redisAsyncFree() will execute pending
+    /* For non-clean disconnects, valkeyAsyncFreeInternal() will execute pending
      * callbacks with a NULL-reply. */
-    if (!(c->flags & REDIS_NO_AUTO_FREE)) {
-      __redisAsyncFree(ac);
+    if (!(c->flags & VALKEY_NO_AUTO_FREE)) {
+      valkeyAsyncFreeInternal(ac);
     }
 }
 
-/* Tries to do a clean disconnect from Redis, meaning it stops new commands
+/* Tries to do a clean disconnect from the server, meaning it stops new commands
  * from being issued, but tries to flush the output buffer and execute
  * callbacks for all remaining replies. When this function is called from a
  * callback, there might be more replies and we can safely defer disconnecting
- * to redisProcessCallbacks(). Otherwise, we can only disconnect immediately
+ * to valkeyProcessCallbacks(). Otherwise, we can only disconnect immediately
  * when there are no pending callbacks. */
-void redisAsyncDisconnect(redisAsyncContext *ac) {
-    redisContext *c = &(ac->c);
-    c->flags |= REDIS_DISCONNECTING;
+void valkeyAsyncDisconnect(valkeyAsyncContext *ac) {
+    valkeyContext *c = &(ac->c);
+    c->flags |= VALKEY_DISCONNECTING;
 
     /** unset the auto-free flag here, because disconnect undoes this */
-    c->flags &= ~REDIS_NO_AUTO_FREE;
-    if (!(c->flags & REDIS_IN_CALLBACK) && ac->replies.head == NULL)
-        __redisAsyncDisconnect(ac);
+    c->flags &= ~VALKEY_NO_AUTO_FREE;
+    if (!(c->flags & VALKEY_IN_CALLBACK) && ac->replies.head == NULL)
+        valkeyAsyncDisconnectInternal(ac);
 }
 
-static int __redisGetSubscribeCallback(redisAsyncContext *ac, redisReply *reply, redisCallback *dstcb) {
-    redisContext *c = &(ac->c);
+static int valkeyGetSubscribeCallback(valkeyAsyncContext *ac, valkeyReply *reply, valkeyCallback *dstcb) {
+    valkeyContext *c = &(ac->c);
     dict *callbacks;
-    redisCallback *cb = NULL;
+    valkeyCallback *cb = NULL;
     dictEntry *de;
     int pvariant;
     char *stype;
     sds sname = NULL;
 
     /* Match reply with the expected format of a pushed message.
      * The type and number of elements (3 to 4) are specified at:
-     * https://redis.io/topics/pubsub#format-of-pushed-messages */
-    if ((reply->type == REDIS_REPLY_ARRAY && !(c->flags & REDIS_SUPPORTS_PUSH) && reply->elements >= 3) ||
-        reply->type == REDIS_REPLY_PUSH) {
-        assert(reply->element[0]->type == REDIS_REPLY_STRING);
+     * https://valkey.io/topics/pubsub#format-of-pushed-messages */
```

Comment:
The currently working link is 
```suggestion
     * https://valkey.io/docs/topics/pubsub/#format-of-pushed-messages */
```
but this might be a moving target..